packages/core/src/contrib/Constraints.ts

```diff
@@ -411,61 +411,25 @@ export const constrDict = {
   },
 
   /**
-   * Make an AABB rectangle disjoint from a vertical line. (Special case of rect-rect disjoint)
+   * Make an AABB rectangle disjoint from a vertical or horizontal line.
+   * (Special case of rect-rect disjoint)
    */
-  disjointRectLineAAVert: (
-    [t1, s1]: [string, any],
-    [t2, s2]: [string, any]
-  ) => {
+  disjointRectLineAA: ([t1, s1]: [string, any], [t2, s2]: [string, any]) => {
     if (!isRectlike(t1) || !isLinelike(t2)) {
       throw Error("expected two line-like shapes");
     }
 
-    const box = bbox(s1.center.contents, s1.w.contents, s1.h.contents);
-    // TODO: Compute the bbox of the line in a nicer way
-    const line = bbox(
-      ops.vdiv(ops.vadd(s2.start.contents, s2.end.contents), constOf(2)),
-      constOf(2),
-      absVal(sub(s2.start.contents[1], s2.end.contents[1]))
-    );
-
-    return ifCond(
-      areDisjointBoxes(box, line),
-      constOf(0),
-      overlap1D(
-        [box.minY, box.maxY],
-        [s2.start.contents[1], s2.end.contents[1]]
-      )
-    );
-  },
-
-  /**
-   * Make an AABB rectangle disjoint from a horizontal line. (Special case of rect-rect disjoint) TODO: Test this
-   */
-  // TODO: Consolidate with disjointRectLineAA; test it
-  disjointRectLineAAHoriz: (
-    [t1, s1]: [string, any],
-    [t2, s2]: [string, any]
-  ) => {
-    if (!isRectlike(t1) || !isLinelike(t2)) {
-      throw Error("expected rect + line-like shapes");
-    }
+    const box = bbox([t1, s1]);
+    const line = bbox([t2, s2]);
 
-    const box = bbox(s1.center.contents, s1.w.contents, s1.h.contents);
-    // TODO: Compute the bbox of the line in a nicer way
-    const line = bbox(
-      ops.vdiv(ops.vadd(s2.start.contents, s2.end.contents), constOf(2)),
-      absVal(sub(s2.start.contents[0], s2.end.contents[0])),
-      constOf(2)
-    );
+    const overlapX = overlap1D([box.minX, box.maxX], [line.minX, line.maxX]);
+    const overlapY = overlap1D([box.minY, box.maxY], [line.minY, line.maxY]);
 
+    // Push away in both X and Y directions
     return ifCond(
       areDisjointBoxes(box, line),
       constOf(0),
-      overlap1D(
-        [box.minX, box.maxX],
-        [s2.start.contents[0], s2.end.contents[0]]
-      )
+      min(overlapX, overlapY)
```

Comment:
Yup this looks like a bug. Tried to figure out why I wrote that, and I actually copied it from the implementation of `disjoint`! Pushing a commit to fix both this function and `disjoint` (which now also incorporates offsets correctly). I'm changing both objectives to `add`, since this allows optimization in both the X and Y directions simultaneously and `overlapX` and `overlapY` are always non-negative by definition.

Reply:
Incidentally this seems to fix the third disjoint test!! I can't seem to get it into the bad local minima it would often fall into before.